runtime/io.c

```diff
@@ -537,27 +595,56 @@ CAMLprim value caml_ml_set_channel_name(value vchannel, value vname)
   return Val_unit;
 }
 
+#define Pair_tag 0
+
+struct channel_list {
+  struct channel* channel;
+  struct channel_list* next;
+};
+
 CAMLprim value caml_ml_out_channels_list (value unit)
 {
   CAMLparam0 ();
   CAMLlocal3 (res, tail, chan);
   struct channel * channel;
+  struct channel_list *channel_list = NULL, *cl_tmp;
+  mlsize_t i, num_channels = 0;
 
-  res = Val_emptylist;
+  caml_plat_lock (&caml_all_opened_channels_mutex);
   for (channel = caml_all_opened_channels;
        channel != NULL;
-       channel = channel->next)
-    /* Include only output channels opened from OCaml and not closed yet.
-       Testing channel->fd >= 0 looks unnecessary, as
+       channel = channel->next) {
+    /* Testing channel->fd >= 0 looks unnecessary, as
        caml_ml_close_channel changes max when setting fd to -1. */
-    if (channel->max == NULL
-        && channel->flags & CHANNEL_FLAG_MANAGED_BY_GC) {
```

Comment:
This condition is missing in multicore, possibly lost in the rebase. Looks like the changes from this PR were not included in multicore: https://github.com/ocaml/ocaml/pull/10136/files. Why? Explains the diff that was observed between `trunk` and `multicore-pr` in `runtime/caml/io.h`.

Reply:
I don't believe this to be intentional. It looks like an oversight.